src/components/select/list-action-button/list-action-button.component.tsx

```diff
@@ -22,6 +31,11 @@ const ListActionButton = React.forwardRef(
         );
       }
 
+      // istanbul ignore if
+      if (!React.isValidElement(listActionButton)) {
```

Comment:
We can't test for this?

Reply:
We certainly could - but we weren't before, and I'm not convinced it adds much value. The coverage issue only came up when converting to TS because, despite the check for truthiness above, a ReactNode can in theory be a string, so TS complains at the `cloneElement` call below.

I'm happy to add a test if you and/or a second reviewer thinks it's worthwhile, but I'm really not convinced.

Reply:
thanks Dan, I looked up the Draggable code that uses `cloneElement` and saw how you did it - it uses `&&`, basically as a substitute for `if`, but it works in such a way as to basically fool the coverage checker (there's nothing "inside" the if so technically all code does get run no matter what) 😆 Happy to copy that here as (as I argued above) I'm really not convinced this needs a test, and it's always nicer not to have to put `istanbul ignore`...

Reply:
actually on reflection this change isn't right, at least not in this place. The issue is that if `listActionButton` is something truthy which is still not a React Element - like a non-empty string - we want to just leave it as is, but the `React.isValidElement(listActionButton) && ...` will end up rendering nothing.

So I'm going to put this back to how I had it, as I think it's the only way to make all possible React Node types behave correctly here while also keeping TS happy. Let me know if you think I should add a test for this case and I will, but for now I'll keep it with the istanbul ignore comment.